packages/voila/src/app.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+// TODO: export from upstream?
+import { createRendermimePlugins } from '@jupyterlab/application/lib/mimerenderers';
```

Comment:
```suggestion
import { createRendermimePlugins } from '@jupyterlab/application';
```

Reply:
https://github.com/jupyterlab/jupyterlab/pull/10117 for reference

Reply:
Yes. This might also need a bump of the `@jupyterlab` packages too (not sure in which `3.1.x` this was released upstream).

We can also update to `4.0.0-alpha.x` directly to grab other fixes as mentioned in https://github.com/voila-dashboards/voila/pull/846#discussion_r777943940.

Reply:
Would you mind if I push new commits to your branch for doing this?

Reply:
Sure, feel free!

Reply:
Thanks!

Actually I am having troubles using jupyterlab's `4.0.0-alpha.x`, as ipywidgets's `@jupyter-widgets/jupyterlab-manager` depends on `^3.0.0` and the tokens are conflicting.

Will it be impossible for me to use both jupyterlab's `4.0.0-alpha.x` and `@jupyter-widgets/jupyterlab-manager` because of this? Do I need to use a local version of `@jupyter-widgets/jupyterlab-manager` that depends on jupyterlab `4.0.0-alpha.x`?

Reply:
Which version of `@jupyterlab/services` is used? Normally it should be `7.0.0-alpha.x`

Reply:
`@voila-dashboards/voila` depends on `@jupyterlab/services: 7.0.0-alpha.2`, but `@jupyter-widgets/base-manager@^1.0.0-beta.1` depends on `@jupyterlab/services: 6.0.0` (sorry my previous comment was missing this information)

Reply:
Ah ok then it is most likely related to this PR: https://github.com/jupyterlab/jupyterlab/pull/11537.

We could try to use `resolutions` to ensure there is a single version of `@jupyterlab/services` being used.